pkg/controllers/management/feature/feature_handler.go

```diff
@@ -94,14 +104,30 @@ func (h *handler) syncHarvesterFeature(obj *v3.Feature) error {
 	return nil
 }
 
-func (h *handler) toggleHarvesterNodeDriver(harvester string) error {
-	if val := features.GetFeatureByName(harvester).Enabled(); val {
-		m, err := h.nodeDriverController.Cache().Get(harvester)
-		if err != nil {
-			return err
+// syncHarvesterNodeDriver ensures that the Harvester node driver is disabled
+// when the Harvester feature is disabled and that the node driver is enabled,
+// when the Harvester feature is enabled - provided that the node driver
+// exists. If it doesn't exist, the node driver is created.
+func (h *handler) syncHarvesterNodeDriver(feat *v3.Feature) error {
+	if feat.Spec.Value == nil {
+		logrus.Debugf("feature %v contains nil value", feat.Name)
+		return nil
+	}
+
+	m, err := h.nodeDriverController.Controller().Lister().Get("", feat.Name)
```

Comment:
```suggestion
	m, err := h.nodeDriverController.Controller().Lister().Get("", datamanagement.HarvesterDriver)
```
I would honestly separate using the name from the feature as the node driver name here.